src/main/scala/com/fulcrumgenomics/bam/OverlappingBasesConsensusCaller.scala

```diff
@@ -76,152 +70,104 @@ class OverlappingBasesConsensusCaller(onlyMaskDisagreements: Boolean = false,
-      if (r1Head.ref < r2Head.ref) {
-        r1BasesBuilder.addAll(r1Bases.slice(from=r1LastReadPos + 1, until=r1Head.read + 1))
-        r1QualsBuilder.addAll(r1Quals.slice(from=r1LastReadPos + 1, until=r1Head.read + 1))
-        r1LastReadPos = r1Head.read
+      if (r1Head.refPos < r2Head.refPos) {
+        r1LastReadPos = r1Head.readPos
         r1Iter.next()
       }
-      else if (r2Head.ref < r1Head.ref) {
-        r2BasesBuilder.addAll(r2Bases.slice(from=r2LastReadPos + 1, until=r2Head.read + 1))
-        r2QualsBuilder.addAll(r2Quals.slice(from=r2LastReadPos + 1, until=r2Head.read + 1))
-        r2LastReadPos = r2Head.read
+      else if (r2Head.refPos < r1Head.refPos) {
+        r2LastReadPos = r2Head.readPos
         r2Iter.next()
       }
-      else { // matched reference bases, so consensus call
-        // add read bases from insertions
-        r1BasesBuilder.addAll(r1Bases.slice(from = r1LastReadPos + 1, until = r1Head.read))
-        r1QualsBuilder.addAll(r1Quals.slice(from = r1LastReadPos + 1, until = r1Head.read))
-        r2BasesBuilder.addAll(r2Bases.slice(from = r2LastReadPos + 1, until = r2Head.read))
-        r2QualsBuilder.addAll(r2Quals.slice(from = r2LastReadPos + 1, until = r2Head.read))
-
-        r1OverlappingBases += 1
-        r2OverlappingBases += 1
-
-        // consensus call current
-        val (base, qual) = consensusCall(
-          base1 = r1Bases(r1Head.read - 1),
-          qual1 = r1Quals(r1Head.read - 1),
-          base2 = r2Bases(r2Head.read - 1),
-          qual2 = r2Quals(r2Head.read - 1)
-        )
-
-        // Only add the consensus call if we aren't masking disagreements or its a no-call
-        if (onlyMaskDisagreements && base != NoCall) {
-          r1BasesBuilder.addOne(r1Bases(r1LastReadPos))
-          r1QualsBuilder.addOne(r1Quals(r1LastReadPos))
-          r2BasesBuilder.addOne(r2Bases(r2LastReadPos))
-          r2QualsBuilder.addOne(r2Quals(r2LastReadPos))
-        } else {
-          r1BasesBuilder.addOne(base)
-          r1QualsBuilder.addOne(qual)
-          r2BasesBuilder.addOne(base)
-          r2QualsBuilder.addOne(qual)
+      else { // both reads are mapped to the same reference bases, so consensus call
+        overlappingBases += 1 // only one for the template
+
+        val base1 = r1.bases(r1Head.readPos - 1)
+        val base2 = r2.bases(r2Head.readPos - 1)
+        if (!SequenceUtil.isNoCall(base1) && !SequenceUtil.isNoCall(base2)) {
+          // consensus call
+          val (newBase1, newQual1, newBase2, newQual2) = consensusCall(
+            base1 = base1,
+            qual1 = r1.quals(r1Head.readPos - 1),
+            base2 = base2,
+            qual2 = r2.quals(r2Head.readPos - 1)
+          )
+
+          if (newBase1 != NoCall && newBase1 != base1) r1Corrected += 1
+          if (newBase2 != NoCall && newBase2 != base2) r2Corrected += 1
+
+          // Only add the consensus call if we aren't masking disagreements or its a no-call
```

Comment:
Maybe kill this comment?  I think `consensusCall` always returns the bases/quals you want to use, and there's no conditional/guard here.